lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1591,20 +1591,10 @@ def __init__(
             if not isinstance(refresh_event, list):
                 refresh_event = [refresh_event]
 
+        self.framework.observe(events.relation_joined, self.set_scrape_job_spec)
         for ev in refresh_event:
             self.framework.observe(ev, self.set_scrape_job_spec)
 
-        # Update relation data every reinit. If instead we used event hooks then observing only
-        # relation-joined would not be sufficient:
-        # - Would need to observe leader-elected, in case there was no leader during
-        #   relation-joined.
```

Comment:
Iiuc, with this change, would need to wait for update-status (is observed btw?) before changes are applied. Is that intentional?

Reply:
Or `refresh_event`, but yes. The other patches squashed this. The charm's unit IP is set in unit databags, and doesn't depend on leader election (there's a potential very racy case where a leader is deposed and a relation is joined during the time that there's no leader at all, but we can make that a separate issue because it isn't affecting current users, and the likelihood is very low)